src/coordinatorapi.cpp

```diff
@@ -369,6 +369,49 @@ void CoordinatorAPI::updateCoordCkptDir(const char *dir)
   _coordinatorSocket.writeAll(dir, strlen(dir) + 1);
 }
 
+void CoordinatorAPI::updateCkptDirGlobally(const char *dir)
+{
+  JASSERT(dir != NULL);
+
+  jalib::JSocket _coordinatorSocket1 = createNewSocketToCoordinator(COORD_ANY);
+  if (!_coordinatorSocket1.isValid()) {
+    return;
+  }
+
+  DmtcpMessage msg(DMT_UPDATE_CKPT_DIR_FOR_ALL_PROCESS);
+  msg.extraBytes = strlen(dir) + 1;
+  _coordinatorSocket1 << msg;
+  _coordinatorSocket1.writeAll(dir, strlen(dir) + 1);
+
+  msg.poison();
+  _coordinatorSocket1 >> msg;
+  msg.assertValid();
+  JASSERT(msg.type == DMT_UPDATE_CKPT_DIR_FOR_ALL_PROCESS_RESULT) (msg.type);
+
+  _coordinatorSocket1.close();
+}
+
+string CoordinatorAPI::getGlobalCkptDir(void)
```

Comment:
I don't have a strong opinion though. If it seems to be a better style and/or if you can anticipate complex string manipulations (for which a C++-style string is obviously better than having to manipulate raw char buffers) we can keep this. The only change I'd suggest in that case is to use the dmtcp alloc arena by switching to the dmtcp namespace, i.e., use: `dmtcp::string CoordinatorAPI::getGlobalCkptDir(void)` instead of: `string CoordinatorAPI::getGlobalCkptDir(void)`.